src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -101,9 +101,9 @@ def prepare_create(self, daemon_spec: CephadmDaemonSpec[AlertManagerSpec]) -> Ce
         # spec should be in spec store
         if not daemon_spec.spec:
             service_name: str = "alertmanager"
-            if service_name in self.mgr.spec_store.specs:
+            if service_name in self.mgr.spec_store:
                 daemon_spec.spec = cast(
-                    AlertManagerSpec, self.mgr.spec_store.specs.get(service_name))
+                    AlertManagerSpec, self.mgr.spec_store[service_name])
```

Comment:
```suggestion
                    AlertManagerSpec, self.mgr.spec_store[service_name].spec)
```
I believe this returns a SpecDescription but we want the actual service spec (same for the other three)